KMeans.java

```diff
@@ -0,0 +1,126 @@
+import java.io.*;
+import java.util.*;
+import java.lang.Math; 
+
+public class KMeans {
+
+	private Vector midpoint (List<Vector> listofvectors) { //computes + returns a midpoint Vector from a given list of Vectors
+		
+		List<Double> mp_coordinates = new ArrayList<Double>();
+		double number_of_vectors = listofvectors.size();
+		double length_of_vector = listofvectors.get(0).numbers.size();	
```

Comment:
what if someone passed a list with 0 vectors. You'd get a array out of bounds error from the `get(0)` call.  So at the top of this function (along with your checks that all vectors of the same length), you should add a check that at least one vector is actually provided